### PR TITLE
docs(samples): add Spanner statement timeout sample

### PR DESCRIPTION
**Pull request by olavloite**
Adds a sample for setting a timeout for a single statement for Spanner.

### CODE REVIEW

File: spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+)
+
+func setStatementTimeout(w io.Writer, db string) error {
+	client, err := spanner.NewClient(context.Background(), db)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	_, err = client.ReadWriteTransaction(context.Background(), func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go:36]: question: what are we throwing away here with the assignment to blank identifier?

Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go:36]: nit: break `func` onto a new line for better scannability.

File: spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		if code == codes.DeadlineExceeded {
+			fmt.Fprintf(w, "Insert statement timed out.\n")
+		} else if code == codes.OK {
+			fmt.Fprintf(w, "%d record(s) inserted.\n", rowCount)
+		} else {
+			fmt.Fprintf(w, "Insert statement failed with error %v\n", err)
+		}
+		return err
+	})
+	return err
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go:56]: nit: I would explicitly check for err of the ReadWriteTransaction, return the err, and then have the func return nil. Same effect as what's written; it's just easier to parse what's happening.

Up to you.

**Review by telpirion — APPROVED** [review]
Very minor nits and questions. Thank you!

File: spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go (hunk 1 above)
Comment (olavloite) [reply spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go:36]: The commit timestamp of the transaction.

Comment (olavloite) [reply spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go:36]: Done

File: spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go (hunk 2 above)
Comment (olavloite) [reply spanner/spanner_snippets/spanner/spanner_set_statement_timeout.go:56]: Done

**Pull request merged**